CMakeLists.txt

```diff
@@ -9,7 +9,7 @@ if(Arrow_FOUND)
   add_definitions(-D_ARROW_EXIST)
 endif()
 
-set(CMAKE_CXX_FLAGS "-std=gnu++17 -ggdb -Wnon-virtual-dtor -Wreorder -Wunused-variable -Wtype-limits -Wsign-compare -Wmaybe-uninitialized")
+set(CMAKE_CXX_FLAGS "-std=gnu++17 -ggdb -Wnon-virtual-dtor -Wreorder -Wunused-variable -Wtype-limits -Wsign-compare -Wmaybe-uninitialized -lre2 -lhs")
```

Comment:
the linkage (-lre2 -lhs) should be set only whether the package is found.
find_package( ... ) --> if ( ... found ) --> add_defintions( ... )

Reply:
Done